art_internal.hpp

```diff
@@ -189,6 +238,12 @@ class [[nodiscard]] basic_node_ptr {
     return reinterpret_cast<T>(tagged_ptr & ptr_bit_mask);
   }
 
+  // same raw_val means same type and same ptr.
```

Comment:
Added a comparator here @laurynas-biveinis

Reply:
The sole caller is `olc_db::iterator::operator ==`, and that one is unused?